src/client.js

```diff
@@ -141,13 +141,34 @@ Client.prototype.connect = function(args) {
 	var webirc = null;
 	var channels = [];
 
-	if (args.join) {
-		var join = args.join.replace(/\,/g, " ").split(/\s+/g);
-		join.forEach(function(chan) {
+	if (args.channels) {
+		var badName = false;
+
+		args.channels.forEach(function(chan) {
+			if (!chan.name) {
+				badName = true;
+				return;
+			}
+
 			channels.push(new Chan({
-				name: chan
+				name: chan.name
 			}));
 		});
+
+		if (badName && client.name) {
+			log.warn("User '" + client.name + "' on network '" + args.name + "' has an invalid channel which has been ignored");
+		}
+	// `join` is kept for backwards compatibility when updating from versions <2.0
+	// also used by the "connect" window
```

Comment:
I don't think we are converting channels on the fly until a new channel is added or removed (and re-ordered, maybe?) so that the list gets dumped in the user configuration. Or are we and I'm missing it?

Depending on the answer, can we log something like `[WARNING] <user> is using the deprecated comma-separated channel list.`? I realize you use the same code for handling old format *and* the join window input though, so not trivial if it stays that way.